src/libs/ck-libs/armci/armci_impl.h

```diff
@@ -240,10 +240,17 @@ class ArmciVirtualProcessor : public TCharmClient1D {
   ~ArmciVirtualProcessor();
  
   pointer BlockMalloc(int bytes) { 
-    return CmiIsomallocContextMalloc(CmiIsomallocGetThreadContext(thread->getThread()), bytes);
+    CmiIsomallocContext * ctx = CmiIsomallocGetThreadContext(thread->getThread());
+    if (ctx == nullptr)
```

Comment:
Should this if-else instead go inside of CmiIsomallocContextFree()? Or is it sometimes not needed

Reply:
It looks like everywhere else already has a check of its own near the call site. threads.C checks `base->isMigratable`, meta_active in memory-isomalloc.C checks `CpvAccess(isomalloc_state).context`. If we want to move these checks to the call site we can discuss that for future cleanup.